flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTaskNetworkInput.java

```diff
@@ -197,6 +205,29 @@ public int getInputIndex() {
 		return checkpointedInputGate.getAvailableFuture();
 	}
 
+	@Override
+	public CompletableFuture<?> prepareSnapshot(long checkpointId) throws IOException {
+		// Note that if considering recovery in future, we should guarantee that the spilled buffers in one channel
+		// should be close together because one record might span multiple buffers.
+		for (int channelIndex = 0; channelIndex < recordDeserializers.length; channelIndex++) {
+			final InputChannel channel = checkpointedInputGate.getChannel(channelIndex);
+
+			recordDeserializers[channelIndex].getUnconsumedBuffer().ifPresent(buffer ->
```

Comment:
How do we know that these data belong to the given checkpoint? Is it because of the single-checkpoint limit?

Reply:
The partial buffer in `RecordDeserializer` should be treated same as the inflight buffers in `RemoteInputChannel` queue. When the given checkpoint is triggered, this partial buffer is also overtaken actually. If there are multiple checkpoints concurrent, this partial buffer might belong to multiple checkpoint states, then that brings the concern of duplicated persistence like incremental checkpoint.

Reply:
> How do we know that these data belong to the given checkpoint? Is it because of the single-checkpoint limit?

Yes, this code (among many other places) only works because we use that assumption for now.

Reply:
Got it, thanks for the explanations.

Does it make sense to add a comment? (and maybe mark somehow so we don't forget it when removing the limit of in-flight checkpoints)